### PR TITLE
Issue521 config from svg scale

### PR DESCRIPTION
**Pull request by Didou09**
Motivation:
-------------
* Provide auto-scaling from user knowledge of 2 points only (see #522 )

Main changes:
----------------
* Autoscaling is now handled
* A unique straight line has to be drawn between 2 points of known real coordinates, the user provides these coordinates as input parameters
* Can also provide the starting point and length of the line
* docstr added to `Config.from_svg()`
* Unit test updated

Example:
-----------

Using data from the unit test
In [1]: import tofu as tf                                                                                                                                                              
/Home/DV226270/ToFu_All/tofu_git/tofu/tofu/imas2tofu/__init__.py:87: UserWarning: 
You do not seem to be using the latest IMAS version:
'module list' vs 'module av IMAS' suggests:
	- Current version: 3.30.0-4.8.4
	- Latest version : 3.32.1-4.9.1-R2019b
  warnings.warn(msg)

In [2]: conf = tf.geom.Config.from_svg('tofu/tests/tests01_geom/test_03_core_data/Inkscape.svg', Name='test', Exp='Test', res=10, point1=(0.7, -2), point2=(2.8, 2))                   
The following structures were loaded:
	- Ves: coil1 (5 pts, None)
	- Ves: coil2 (6 pts, None)
	- Ves: coil3 (5 pts, None)
	- Ves: vesselInner (21 pts, None)
	- Ves: vesselOuter (11 pts, None)
from /Home/DV226270/ToFu_All/tofu_git/tofu/tofu/tests/tests01_geom/test_03_core_data/Inkscape.svg
/Home/DV226270/ToFu_All/tofu_git/tofu/tofu/geom/_core.py:481: UserWarning: Struct instance: double identical points in Poly
  => 1 points removed
  => Poly goes from 11 to 10 points
  warnings.warn(msg)
/Home/DV226270/ToFu_All/tofu_git/tofu/tofu/geom/_core.py:481: UserWarning: Struct instance: double identical points in Poly
  => 1 points removed
  => Poly goes from 5 to 4 points
  warnings.warn(msg)
/Home/DV226270/ToFu_All/tofu_git/tofu/tofu/geom/_core.py:481: UserWarning: Struct instance: double identical points in Poly
  => 1 points removed
  => Poly goes from 6 to 5 points
  warnings.warn(msg)
/Home/DV226270/ToFu_All/tofu_git/tofu/tofu/geom/_core.py:481: UserWarning: Struct instance: double identical points in Poly
  => 1 points removed
  => Poly goes from 21 to 20 points
  warnings.warn(msg)

In [3]: conf.plot()                                                                                                                                                                    
Out[3]: 
[<matplotlib.axes._axes.Axes at 0x7fc57ff81d50>,
 <matplotlib.axes._axes.Axes at 0x7fc57f5cd610>]

image
image

### CODE REVIEW

**Comment by Didou09** [comment]
@lasofivec , I'm leaving with the bus,

If you agree and validate, can you:
- merge
- delete the branch
- Redo the video using this brand new feature for auto-scaling ?
- Update the doc with the video

Thanks !

File: tofu/geom/_comp.py
Comment (lasofivec) [inline tofu/geom/_comp.py:56]: what do you expect to have in the first `{ }`? maybe `var`? so something along the lines:
```
    "Arg {} must be a float!\n".format(var)
    + "Provided: {}".format(type(var))
```

File: tofu/geom/_comp.py
Comment (lasofivec) [inline tofu/geom/_comp.py:196]: if I understand correctly, you will check all paths and see if any are references, if so, you keep the poly to compute automatic scaling and delete from the configuration? And how do you know if there are? just because there is a "z" in the poly?

File: tofu/geom/_core.py
Comment (lasofivec) [inline tofu/geom/_core.py:3867]: tbh, I feel like `point1` and `point2` are not specific enough names... `point_ref1/2`? `strart_ref/end_ref`? or even `reference1/2`? what do you think?

File: tofu/geom/_core.py
Comment (lasofivec) [inline tofu/geom/_core.py:3883]: "two points straight line"? or can it be more than 2 points?

File: tofu/geom/_core.py
Comment (lasofivec) [inline tofu/geom/_core.py:3894]: The*

File: tofu/geom/_core.py
Comment (lasofivec) [inline tofu/geom/_core.py:3908]: I have forgotten the Name argument, and there is a long stack on the trace callback, maybe they shouldn't be optional

**Review by lasofivec — COMMENTED** [review]
I will work from this branch for the video, but I wont merge it as is

**Comment by lasofivec** [comment]
This is the best one yet:

https://youtu.be/MFwMZL7JjhI

There are some other trials that I don't think are as good, you now have access to the youtube account so you can see them all (they are unlisted), for example:

- https://youtu.be/tuz9jczmjjs

**Comment by lasofivec** [comment]
I think everything is done except for merge, since I think the changes I made are worth doing before merging PR

File: tofu/geom/_comp.py
Comment (Didou09) [reply tofu/geom/_comp.py:51]: It's the name of the variable, this is a generic abstract function to check that a variable is of type float or int and raise an automated error message otherwise

Comment (Didou09) [reply tofu/geom/_comp.py:51]: and the value to a default value if None

File: tofu/geom/_comp.py
Comment (Didou09) [reply tofu/geom/_comp.py:196]: Yes, in svg formatting, a ''z at the end of the str representation of a path means it is closed. If there is only one path with no z it is a unclosed path so it is the straight line

File: tofu/geom/_core.py
Comment (Didou09) [reply tofu/geom/_core.py:3867]: done

File: tofu/geom/_core.py
Comment (Didou09) [reply tofu/geom/_core.py:3883]: done

File: tofu/geom/_core.py
Comment (Didou09) [reply tofu/geom/_core.py:3894]: done

File: tofu/geom/_core.py
Comment (Didou09) [reply tofu/geom/_core.py:3908]: The docstr says must, so I guess it is clear, but I agree this can be annoying.
In a future version I will try make this optional or set default values when they are forgotten, but this may take some time so I suggest we keep it as it is for now

**Review by lasofivec — APPROVED** [review]

**Comment by Didou09** [comment]
Good, preparing the PR for 1.4.11 (devel-> master), finishing release notes

**Pull request merged**
